app/parse.py

```diff
@@ -8,5 +14,31 @@ class Course:
     duration: str
 
 
+def parse_single_course(soup: Tag) -> Course:
+    return (Course(
+        name=soup.select_one("a.ProfessionCard_title__fEWio > h3").text,
+        short_description=soup.select_one(
+            ".ProfessionCard_professionTags__2iarD + "
+            "p.typography_landingTextMain__Rc8BD"
+        ).text,
+        duration=soup.select_one(
+            ".ProfessionCardTags_regularTag__dqOGj:last-child span"
+        ).text
+    ))
+
+
 def get_all_courses() -> list[Course]:
-    pass
+    try:
+        text = requests.get(BASE_URL).content
+    except requests.exceptions.RequestException as e:
+        print(e)
+    soup = BeautifulSoup(text, "html.parser")
+    courses = soup.select(".ProfessionCard_cardWrapper__2Q8_V")
+    return [parse_single_course(course) for course in courses]
```

Comment:
1. **Exception Handling**: The `except` block only prints the exception. It would be better to handle the exception more gracefully, such as logging the error or retrying the request.
2. **Variable Scope**: If an exception occurs, the `text` variable will not be defined, leading to a `NameError`. It would be better to set a default value for `text` or handle this case.